### PR TITLE
21695: Fixes WASM debug build, packaging updates

### PR DESCRIPTION
**Pull request by apbassett**
* Fixes issue where WASM debug build did not actually have debug information available
* Adds `web` to `sENVIRONMENTS` linker flag
* Removes quiet mode from CI/CD build step for future debugability
* Splits build process into separate release/debug phases for WASM instead of shoehorning a debug build into the release build

### CODE REVIEW

File: build/cmake/global_compiler_flags.cmake [hunk 1]
```diff
@@ -75,7 +75,7 @@ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" S
...
+        string(APPEND CMAKE_EXE_LINKER_FLAGS " -sINVOKE_RUN=0 -sALLOW_MEMORY_GROWTH=1 -sINITIAL_MEMORY=65536000 -sMEMORY_GROWTH_GEOMETRIC_STEP=0.50 -sMODULARIZE=1 -sEXPORT_NAME=AmalgamRuntime -sENVIRONMENT=worker,node,web -sEXPORTED_RUNTIME_METHODS=cwrap,ccall,FS,setValue,getValue,UTF8ToString -sEXPORTED_FUNCTIONS=_malloc,_free,_LoadEntity,_CloneEntity,_VerifyEntity,_StoreEntity,_ExecuteEntity,_ExecuteEntityJsonPtr,_DestroyEntity,_GetEntities,_SetRandomSeed,_SetJSONToLabel,_GetJSONPtrFromLabel,_SetSBFDataStoreEnabled,_IsSBFDataStoreEnabled,_GetVersionString,_SetMaxNumThreads,_GetMaxNumThreads,_GetConcurrencyTypeString,_DeleteString --preload-file /wasm/tzdata@/tzdata --preload-file /wasm/etc@/etc")
```
Comment (fulpm) [inline build/cmake/global_compiler_flags.cmake]: Why was `web` added to the environments?

Comment (apbassett) [reply build/cmake/global_compiler_flags.cmake]: @lancegliser noticed a log from Node that was complaining about it not being present (using the debug build). If you think there's a reason to not add this for both release and debug, do let me know!

Comment (fulpm) [reply build/cmake/global_compiler_flags.cmake]: Looks like the diff between on/off is negligible, should be fine to leave for both builds

File: build/cmake/global_compiler_flags.cmake [hunk 2]
```diff
@@ -139,6 +139,14 @@ if(IS_MSVC)
     add_compile_definitions(UNICODE _UNICODE)
 endif()
 
+# Ensure that debug WASM builds include additional WASM-specific linker flags 
+if (IS_WASM AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS " -sASSERTIONS=2 -sSAFE_HEAP -sDEMANGLE_SUPPORT")
```
Comment (lancegliser) [inline build/cmake/global_compiler_flags.cmake]:
```suggestion
    string(APPEND CMAKE_EXE_LINKER_FLAGS " -sASSERTIONS=2 -sDEMANGLE_SUPPORT")
```

We do some custom alignment work if I heard Chris right this morning. Explains why I had to take this out to even boot the debug wasm.

Comment (lancegliser) [inline build/cmake/global_compiler_flags.cmake]:
```suggestion
    string(APPEND CMAKE_EXE_LINKER_FLAGS " -sASSERTIONS=2 -sSAFE_HEAP")
```

> em++: warning: DEMANGLE_SUPPORT is deprecated (mangled names no longer appear in stack traces). Please open a bug if you have a continuing need for this setting [-Wdeprecated]

Looks like they started reporting unmangled names by default back in Feb or March.

File: build/cmake/global_compiler_flags.cmake [hunk 3]
```diff
@@ -75,7 +75,7 @@ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" S
...
+        string(APPEND CMAKE_EXE_LINKER_FLAGS " -sINVOKE_RUN=0 -sALLOW_MEMORY_GROWTH=1 -sINITIAL_HEAP=16777216 -sSTACK_SIZE=8388608 -sMEMORY_GROWTH_GEOMETRIC_STEP=0.50 -sMODULARIZE=1 -sEXPORT_NAME=AmalgamRuntime -sENVIRONMENT=worker,node,web -sEXPORTED_RUNTIME_METHODS=cwrap,ccall,FS,setValue,getValue,UTF8ToString -sEXPORTED_FUNCTIONS=_malloc,_free,_LoadEntity,_CloneEntity,_VerifyEntity,_StoreEntity,_ExecuteEntity,_ExecuteEntityJsonPtr,_DestroyEntity,_GetEntities,_SetRandomSeed,_SetJSONToLabel,_GetJSONPtrFromLabel,_SetSBFDataStoreEnabled,_IsSBFDataStoreEnabled,_GetVersionString,_SetMaxNumThreads,_GetMaxNumThreads,_GetConcurrencyTypeString,_DeleteString --preload-file /Users/lance.gliser/Projects/github-howsoai/amalgam/local/wasm/tzdata@/tzdata --preload-file /Users/lance.gliser/Projects/github-howsoai/amalgam/local/wasm/etc@/etc")
```
Comment (lancegliser) [inline build/cmake/global_compiler_flags.cmake]: Probably bad to have Lance's machine.

Comment (lancegliser) [reply build/cmake/global_compiler_flags.cmake]: Removed

File: build/cmake/global_compiler_flags.cmake [hunk 4]
```diff
@@ -75,7 +75,7 @@ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" S
...
+        string(APPEND CMAKE_EXE_LINKER_FLAGS " -sINVOKE_RUN=0 -sALLOW_MEMORY_GROWTH=1 -sINITIAL_HEAP=65536000 -sSTACK_SIZE=33554432 -sMEMORY_GROWTH_GEOMETRIC_STEP=0.50 -sMODULARIZE=1 -sEXPORT_NAME=AmalgamRuntime -sENVIRONMENT=worker,node,web -sEXPORTED_RUNTIME_METHODS=cwrap,ccall,FS,setValue,getValue,UTF8ToString -sEXPORTED_FUNCTIONS=_malloc,_free,_LoadEntity,_CloneEntity,_VerifyEntity,_StoreEntity,_ExecuteEntity,_ExecuteEntityJsonPtr,_DestroyEntity,_GetEntities,_SetRandomSeed,_SetJSONToLabel,_GetJSONPtrFromLabel,_SetSBFDataStoreEnabled,_IsSBFDataStoreEnabled,_GetVersionString,_SetMaxNumThreads,_GetMaxNumThreads,_GetConcurrencyTypeString,_DeleteString --preload-file /wasm/tzdata@/tzdata --preload-file /wasm/etc@/etc")
```
Comment (fulpm) [inline build/cmake/global_compiler_flags.cmake]: INITIAL_HEAP and STACK_SIZE should not start this high on non-debug builds `-sINITIAL_HEAP=16777216 -sSTACK_SIZE=8388608` should be reasonable for production builds

Comment (fulpm) [reply build/cmake/global_compiler_flags.cmake]: Could also maybe do 32mb on the initial heap

**Review by fulpm — APPROVED** [review]

**Pull request merged**